models/topics.model.js

```diff
@@ -23,10 +23,17 @@ exports.getArticleDataById = (id) => {
     });
 };
 
-exports.getArticlesData = () => {
+exports.getArticlesData = (topic) => {
+  let articlePromise = topic
+    ? db.query(
+        "SELECT * FROM articles WHERE topic = $1 ORDER BY created_at DESC",
+        [topic]
+      )
+    : db.query("SELECT * FROM articles ORDER BY created_at DESC");
+
   return Promise.all([
     db.query("SELECT article_id FROM comments"),
-    db.query("SELECT * FROM articles ORDER BY created_at DESC"),
+    articlePromise,
```

Comment:
Interesting use of a ternary operator to determine which query you want to use - another way to go about solving this problem would be to construct your query string depending on whether you get a topic query